vowpalwabbit/cats_tree.cc

```diff
@@ -360,16 +361,18 @@ base_learner* setup(options_i& options, vw& all)
     options.replace("link", "glf1");
   }
 
-  auto tree = scoped_calloc_or_throw<cats_tree>();
+  auto tree = VW::make_unique<cats_tree>();
```

Comment:
It looks like `_cost_star, _a, _b` are unitialized after this call.

Reply:
Just recommitted. Let me know if manually initializing _a and _b is preferred to creating a constructor in node_cost.

Reply:
Having the node_cost initialization done internally is good as it affects any other usage too. So this approach is better